checks/compile_gcc.go

```diff
@@ -0,0 +1,113 @@
+package checks
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+)
+
+func init() {
+	availableChecks["compile-gcc"] = func(args map[string]interface{}) (Checker, error) {
+		return CompileGcc{}.FromArgs(args)
+	}
+}
+
+// CompileGcc runs gcc compile.
+//
+// Type:
+//	 - compile-gcc
+//
+// Support Platforms:
+//   - Mac
+//   - Linux
+//   - Windows
+//
+// Arguments:
+//   source (required): The source code of the script.
+//   interpreter: path to bash. Default is 'gcc' from the PATH
+//				  but it will also check if gcc exists in toolchain folders.
+type CompileGcc struct {
+	Source      string
+	Interpreter string
+	Cflags      string
+}
+
+// Check Runs a gcc command and checks the return code
+func (cg CompileGcc) Check() error {
+
+	tmpfolder, err := ioutil.TempDir("", "compileGcc_")
+	if err != nil {
+		return fmt.Errorf("Problem creating a tmpdir: %s", err)
+	}
+
+	srcfileName := tmpfolder + "/src.c"
+	outfileName := tmpfolder + "/out.o"
+
+	srcfile, err := os.Create(srcfileName)
+	defer os.RemoveAll(tmpfolder)
```

Comment:
Should go right after the error checking of ioutil.TmpDir so it is close to the code it's related to.